### PR TITLE
Detect and error out on RFun refinements instead of discarding them

### PR DESCRIPTION
**Pull request by spinda**
Fixes the rest of #292.

Tests *should* pass, but please wait for them to finish before merging.

### CODE REVIEW

**Comment by spinda** [comment]
Tests are passing except for `icfp_pos` and `icfp_neg`, but those were [already failing on `master`](https://travis-ci.org/ucsd-progsys/liquidhaskell/builds/56820549) before this PR.

**Comment by nikivazou** [comment]
I am working on these tests right now

**Comment by spinda** [comment]
Were you able to fix those tests?

**Comment by nikivazou** [comment]
I think so, Travis is acting weird... They are fixed locally but fail most Travis builds....

File: src/Language/Haskell/Liquid/Bare/Check.hs [hunk 1]
```diff
@@ -25,12 +25,12 @@ import Text.PrettyPrint.HughesPJ
...
 import qualified Data.HashMap.Strict as M
 
-import Language.Fixpoint.Misc (applyNonNull, group, mapSnd, snd3, errorstar, safeHead)
+import Language.Fixpoint.Misc (applyNonNull, group, mapSnd, errorstar, safeHead)
 import Language.Fixpoint.Sort (checkSorted, checkSortedReftFull, checkSortFull)
 import Language.Fixpoint.Types hiding (R)
 
-import Language.Haskell.Liquid.GhcMisc (realTcArity, showPpr, sourcePosSrcSpan)
-import Language.Haskell.Liquid.Misc (dropThd3, firstDuplicate)
+import Language.Haskell.Liquid.GhcMisc (realTcArity, showPpr, showPpr, sourcePosSrcSpan)
```
Comment (nikivazou) [inline src/Language/Haskell/Liquid/Bare/Check.hs]: delete extra showPpr

**Comment by spinda** [comment]
Travis build is here:

https://travis-ci.org/spinda/liquidhaskell/builds/57731440

The only thing failing is WhileM, which is failing on `master`. ByteString's Char8 and Lazy/Char8 are also failing on `master`, but for this build I merged with a version a few commits back before that failure was introduced.

**Comment by nikivazou** [comment]
I pushed a fix for bytestring on master. Can you rebase?

**Comment by spinda** [comment]
@nikivazou Done! Tests are re-running now.

**Comment by spinda** [comment]
Everything's passing, except for the `WhileM` getting killed as we were discussing in #378.

**Comment by spinda** [comment]
`WhileM` passes when I run it locally, though, so everything should be good.

**Comment by nikivazou** [comment]
Nice! 

But, how did you make Travis accept `whileM`?

**Comment by spinda** [comment]
I just restarted the build until it didn't error out... I'm pretty sure the
issue is that we're running out of memory in the Travis VM.

**Comment by nikivazou** [comment]
This is strange tough, as I would expect for example `bytestring

**Pull request merged**